modules/pam_timestamp/pam_timestamp.c

```diff
@@ -487,8 +502,18 @@ pam_sm_authenticate(pam_handle_t *pamh, int flags, int argc, const char **argv)
 		message_end = message + strlen(path) + 1 + sizeof(then);
 
 		/* Regenerate the MAC. */
-		hmac_sha1_generate_file(pamh, &mac, &maclen, TIMESTAMPKEY, 0, 0,
-					message, message_end - message);
+#ifdef WITH_OPENSSL
+		if (hmac_generate(pamh, debug, &mac, &maclen, TIMESTAMPKEY,
+				  ROOT_USER, ROOT_GROUP, message, message_end - message)) {
+			pam_syslog(pamh, LOG_ERR, "Error generating hmac");
```

Comment:
There should be already a syslog message from hmac_generate, shouldn't it? If so I'd drop this to not spam the syslog too much.